### ISSUE TITLE
Config option for the in-game sound volume

### ISSUE THREAD

**Issue by ivan-mogilko** [adventuregamestudio/ags#944]
This may be wanted to have a config option(s) to adjust in-game volume:
* to complement and sync with the in-game volume controls;
* to provide volume setting in case developer did not implement any in-game controls.

The trivial choice is to have master volume setting, corresponding to System.Volume property.

Individual settings per each audio type may be less trivial, because audio types can be customized in game. But this still is worth considering.

**Comment by messengerbag** [comment]
I don't think this is a good idea. A game may have good reason to force certain volume settings (volume off, most obviously), and may rely on logic that assumes only script can change the volume.

And I don't think there is a real player need for this feature: to control the master volume the OS volume controls (or your speaker volume) is the way to go. This only complicates and clutters up winsetup.

**Comment by morganwillcock** [comment]
I'd probably agree, it raises a lot of questions and the implementation can't take into account the full path the audio takes before someone hears it. I'd mainly worry you would also get someone who shipped the game with a config that had the audio turned down (say to 50%) and still didn't implement the volume controls, it then creates the opposite problem.

Perhaps a better solution is to ensure that the new game templates have some keyboard shortcuts that step System.Volume up and down. If the developer forgot to implement their own volume controls and knowingly deleted the existing shortcuts then I don't think you can do much more for them.

**Comment by ivan-mogilko** [comment]
Maybe did not thought this through before posting.

I think instead I should have suggested an engine's own layer of control which could be activated as an emergency when game creator did not provide some settings.
